cmd/uhc/cluster/describe/describe.go

```diff
@@ -108,26 +125,76 @@ func run(cmd *cobra.Command, argv []string) {
 	clusterResource := resource.Cluster(argv[0])
 
 	// Retrieve the collection of clusters:
-	response, err := clusterResource.Get().
-		Send()
+	response, err := clusterResource.Get().Send()
+
+	// Get cluster body
+	cluster := response.Body()
+
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Can't retrieve clusters: %s", err)
 		os.Exit(1)
 	}
 
-	cluster := response.Body()
-	fmt.Printf("ID:       %s\n"+
-		"Name:     %s.%s\n"+
-		"Masters:  %d\n"+
-		"Computes: %d\n"+
-		"Region:   %s\n"+
-		"Multi-az: %t\n",
-		cluster.ID(),
-		cluster.Name(),
-		cluster.DNS().BaseDomain(),
-		cluster.Nodes().Master(),
-		cluster.Nodes().Compute(),
-		cluster.Region().ID(),
-		cluster.MultiAZ(),
-	)
+	if args.output {
+		// Create a filename based on cluster name
+		filename := ("cluster-" + cluster.ID() + ".json")
+		// Attempt to create file
+		myFile, err := os.Create(filename)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to create file: %v", err)
+			os.Exit(1)
+		}
+		// Reasign encoder io.Writer to file writer
+		encoder := json.NewEncoder(myFile)
+		encoder.SetIndent("", " ")
+		// Dump encoder content into file
+		err = v1.MarshalCluster(cluster, encoder)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to Marshal cluster into file: %v", err)
+			os.Exit(1)
+		}
+	}
+	// Get full Api Response (JSON)
```

Comment:
Add a blank line before the comment, end it with punctuation, and be careful with case.

I am a bit picky with this, the result of typing lessons I took long long time ago: `// Get full API response (JSON):`